internal/kms/job/data_key_version_destruction_monitor_job.go

```diff
@@ -0,0 +1,60 @@
+package job
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/scheduler"
+)
+
+type dataKeyVersionDestructionMonitorJob struct {
+	kmsRepo *kms.Kms
+}
+
+func newDataKeyVersionDestructionMonitorJob(ctx context.Context, kmsRepo *kms.Kms) (*dataKeyVersionDestructionMonitorJob, error) {
```

Comment:
I don't know that it's worth testing this, it's all a very thin wrapper around `MonitorDataKeyVersionDestruction`. I added a very basic error around this constructor but let me know if you want me to add more.

Reply:
I agree it's currently a thin wrapper... but should we just skip testing all the simple stuff or should the code base have discernible patterns and conventions?  If we get an PR from a contributor outside the team with no constructor tests, how do we push back when they point out all the things we just don't bother to test?  The same goes for onboarding new devs.  Happy to discuss further.


BTW, I would add a test for invalid parameters as well.

Reply:
I think the test I added satisfies this (it does test for invalid parameters). I agree that conventions are important, I do think we err on the side of too many tests rather than too few, from what I have seen, but I'm happy to follow convention 😄.